featuretools/primitives/standard/aggregation_primitives.py

```diff
@@ -343,7 +343,11 @@ def __init__(self, n=3):
 
     def get_function(self, agg_type=Library.PANDAS):
         def n_most_common(x):
-            array = np.array(x.value_counts().index[:self.n])
+            # Counts of 0 remain in value_counts output if dtype is category
+            # so we need to remove them
+            counts = x.value_counts()
+            counts = counts[counts > 0]
+            array = np.array(counts.index[:self.n])
```

Comment:
I think it'd be good to add a test case to confirm this behavior of the primitive. Probably out of scope for this PR, but can you create an issue to track if you decide not to add?

Reply:
Do we also want to add this logic add here?

https://github.com/alteryx/featuretools/blob/be7de80e0a28d4797eaa186f57c19850674cc0df/featuretools/tests/primitive_tests/test_direct_features.py#L138-L143

Reply:
Alternatively, could we subclass `NMostCommon` to avoid duplicating this code?

```python
class ThreeMostCommonCat(NMostCommon):
    ...
```

Reply:
@jeff-hernandez `NMostCommon` is used in quite a few different tests, perhaps most directly in `test_feature_set_calculator.py::test_topn`.  It is also tested to some degree by the docstring test in the primitive clas definition. Do you think we need to test more than that?

Also, good call on updating the code in test_direct_features.py. I made the update there as well, so that test now passes.

Reply:
It seems like we're adding new behavior to `NMostCommon`.  Are we currently testing this new behavior? If not, I think adding a new test or even modifying an existing one would make sense (in a separate PR).

Reply:
Just to be clear, we aren't actually adding new behavior here - just fixing what I think was incorrect behavior in the current code for some inputs that have a categorical dtype. The previous implementation for this primitive would return items that were not present in the series, as long as they were present in the categories values of the category dtype. I don't think this situation was being tested before, but now that we are setting more columns to have category dtypes with Woodwork it is.

We can add a new test to more explicitly test this situation though so it is clear how we expect this primitive to behave.

Since the code it being changed here in order for our CFM tests to pass, I think it probably makes sense if we add the additional primitive test along with the code change.

Reply:
If we aren't testing the current behavior, let's add an explicit test and/or fix the tests that tested the previous behavior. 

> Since the code it being changed here in order for our CFM tests to pass, I think it probably makes sense if we add the additional primitive test along with the code change.

Sounds good!

Reply:
Added a new test in `test_agg_primitives.py` called `test_nmostcommon_categorical` to cover this situation.